pkg/sources/docker/docker_caching.go

```diff
@@ -0,0 +1,93 @@
+package docker
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// ConnectToLayersDB opens the SQLite database and returns a connection.
+func ConnectToLayersDB(dbName string) (*sql.DB, error) {
+	conn, err := sql.Open("sqlite3", dbName)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to database: %w", err)
+	}
+	return conn, nil
+}
+
+// InitializeLayersDB initializes the SQLite database with the digest table
+// Schema: digest (digest TEXT UNIQUE, verified BOOLEAN, unverified_with_error BOOLEAN, completed BOOLEAN)
+// It returns an error if encountered
+func InitializeLayersDB(db *sql.DB) error {
```

Comment:
@rosecodym brought up a similar point. It _might_ make sense to consolidate to one. I was just considering the situation where there's an unverified secret due to something like a network error. In that case, we wouldn't want to skip that layer, in case that secret is actually valid. The reason for splitting into two vars is I could image an org blocks certain traffic that keeps causing a verification error and the user doesn't want to have to keep scanning that layer and would rather only look at verified secrets. But that could be a super small edge case and not worth the added code complexity.